### DISCUSSION TITLE
Determining what theme to use

### DISCUSSION THREAD

**Discussion by wf5w** [sharkdp/bat#2284] [Show and tell]
I got tired of trying random themes to make most of my files work, still not all do, but I now have most files working good with the theme I chose. So how did I settle on a theme? I wrote this script, and tested it with different files to see which themes worked best.

> #!/usr/bin/env bash
> 
> if [ -z "$1" ]; then
>>   echo usage: dobatthemes filename
>>   exit 1
> fi
> 
> bat --list-themes > bat_themes.txt
> 
> while read t; do
>>   echo theme $t; sleep 4
>>   bat --theme "$t" "$1"
> done < bat_themes.txt
> 
> rm bat_themes.txt
>

**Comment by sharkdp** [comment]
See also the suggestion in this README section: https://github.com/sharkdp/bat#highlighting-theme
